pkg/pgmodel/model/interface.go

```diff
@@ -49,7 +49,7 @@ func MetricTableName(conn pgxconn.PgxConn, metric string) (string, bool, error)
 	var possiblyNew bool
 	defer res.Close()
 	if !res.Next() {
-		if res.Err() != nil {
+		if err := res.Err(); err != nil {
```

Comment:
Can we check this before this block and if no error, then verify if metric_name is empty. if yes, send a metric not exists error by reusing some previously declared error.

Reply:
no res.Error can change with res.Next and res.Scan calls